pkg/parse.go

```diff
@@ -42,67 +44,71 @@ func NewParser(buildTags []string) *Parser {
 	}
 }
 
-func (p *Parser) Parse(ctx context.Context, path string) error {
-	// To support relative paths to mock targets w/ vendor deps, we need to provide eventual
-	// calls to build.Context.Import with an absolute path. It needs to be absolute because
-	// Import will only find the vendor directory if our target path for parsing is under
-	// a "root" (GOROOT or a GOPATH). Only absolute paths will pass the prefix-based validation.
-	//
-	// For example, if our parse target is "./ifaces", Import will check if any "roots" are a
-	// prefix of "ifaces" and decide to skip the vendor search.
-	path, err := filepath.Abs(path)
-	if err != nil {
-		return err
-	}
+func (p *Parser) Parse(ctx context.Context, pattern string) error {
+	log := zerolog.Ctx(ctx)
 
-	dir := filepath.Dir(path)
-
-	files, err := ioutil.ReadDir(dir)
-	if err != nil {
+	info, err := os.Stat(pattern)
+	if err != nil && !os.IsNotExist(err) {
 		return err
 	}
 
-	for _, fi := range files {
-		log := zerolog.Ctx(ctx).With().
-			Str(logging.LogKeyDir, dir).
-			Str(logging.LogKeyFile, fi.Name()).
-			Logger()
-		ctx = log.WithContext(ctx)
-
-		if filepath.Ext(fi.Name()) != ".go" || strings.HasSuffix(fi.Name(), "_test.go") {
-			continue
+	var query string
+	switch {
+	case os.IsNotExist(err):
+		// The pattern represents one or more packages and should be passed directly to the package loader.
+		log.Debug().Msgf("Loading packages corresponding to pattern %q.", pattern)
+		query = pattern
+
+	case !info.IsDir():
+		// A file should be passed directly to the package loader as a 'file' query.
+		if filepath.Ext(pattern) != ".go" {
+			return fmt.Errorf("specified file %q cannot be parsed as it is not a source file", pattern)
+		} else if strings.HasPrefix(info.Name(), ".") || strings.HasPrefix(info.Name(), "_") {
```

Comment:
Is this not different behavior? I may be wrong but I dont' see anywhere where mockery would have skipped `.*` or `_*` files.

Reply:
This is some [existing logic](https://github.com/vektra/mockery/blob/b4bca183629684bf40b069cc339ebc22015ded71/pkg/walker.go#L75-L77) that I took from `Walker` and moved here. I am not partial to keeping or removing it, I just moved to keep the existing semantics. :slightly_smiling_face: 

Technically it would be a breaking change to remove this.